### PR TITLE
DEPR: DataFrame.groupby(axis=1)

### LINKED ISSUE #51203: DEPR: DataFrame.groupby(axis=1)
It isn't obvious to me what you can do with `df.groupby(axis=1)` that you can't do with `df.T.groupby(axis=0)`.  In many cases we do the transpose under the hood anyway, so it isn't saving much/anything perf-wise.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #51203 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/user_guide/categorical.rst [hunk 1]
```diff
@@ -625,8 +625,8 @@ even if some categories are not present in the data:
     df = pd.DataFrame(
         data=[[1, 2, 3], [4, 5, 6]],
         columns=pd.MultiIndex.from_arrays([["A", "B", "B"], columns]),
-    )
-    df.groupby(axis=1, level=1).sum()
+    ).T
+    df.groupby(level=1).sum()
```
Comment (rhshadrach) [inline doc/source/user_guide/categorical.rst:629]: I think this whole example can also be removed; it started as `DataFrame.sum` with `level=1` passed which used groupby under the hood, making the unused categories show up (observed=False is the default). Now `level` is not an argument do `DataFrame.sum`, and so the docs here are not correct - unused categories don't show up in `DataFrame.sum`.

Groupby is shown in the next example, which I think is sufficient.

But potentially removing this example would also be okay for a different PR.

**Review by rhshadrach — COMMENTED** [review]
lgtm, just a thought about the docs. Also, are we good with targeting this for 2.1?

File: doc/source/user_guide/categorical.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/user_guide/categorical.rst:629]: > But potentially removing this example would also be okay for a different PR.

I'd like to punt on this for now, yah

**Review by rhshadrach — APPROVED** [review]
lgtm; assuming no opposition is voiced to the deprecation I plan to merge this Tuesday

**Comment by jbrockmendel** [comment]
Any more thoughts here?

**Comment by rhshadrach** [comment]
@jbrockmendel - thanks for the ping; just a conflict with the whatsnew.

**Comment by rhshadrach** [comment]
Thanks @jbrockmendel

**Comment by cfaaron** [comment]
nb  the 2.1.0 docs at https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html do not mention this depreciation

**Comment by rhshadrach** [comment]
Thanks @cfaaron - I've opened #54873.

**Pull request merged**